core/src/main/java/tc/oc/pgm/consumable/ConsumableMatchModule.java

```diff
@@ -0,0 +1,63 @@
+package tc.oc.pgm.consumable;
+
+import com.google.common.collect.ImmutableSet;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerItemConsumeEvent;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.ParticipantState;
+import tc.oc.pgm.kits.tag.ItemTags;
+
+public class ConsumableMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+  private final ImmutableSet<ConsumableDefinition> consumableDefinitions;
+
+  public ConsumableMatchModule(
+      Match match, ImmutableSet<ConsumableDefinition> consumableDefinitions) {
+    this.match = match;
+    this.consumableDefinitions = consumableDefinitions;
+  }
+
+  @EventHandler
+  private void onItemConsume(PlayerItemConsumeEvent event) {
+    ConsumableDefinition consumableDefinition = getConsumableDefinition(event.getItem());
+    if (consumableDefinition == null) return;
+    if (consumableDefinition.getCause() != ConsumeCause.EAT) return;
+
+    Player player = event.getPlayer();
+    MatchPlayer matchPlayer = match.getPlayer(player);
+    ParticipantState playerState = match.getParticipantState(player);
+    if (playerState == null) return;
+
+    if (consumableDefinition.getPreventDefault()) {
+      ItemStack itemInHand = player.getItemInHand();
+      if (itemInHand.getAmount() > 1) {
+        itemInHand.setAmount(itemInHand.getAmount() - 1);
+      } else {
+        player.setItemInHand(null);
+      }
```

Comment:
I've seen this block of code in a few places, potentially look into moving to some util?

Reply:
Not sure how utils is structured — should I add this to `util/Players.java`, make a new util, or something else?

Reply:
I'd probably stick it in some inventory utils or similar, like `InventoryUtils.consumeItem(player)`

Reply:
i double-checked and it seems like `InventoryUtils` already exists, and the items are very related (ie: place stack, or similar), so it'd fit right in as `InventoryUtils.consumeItem(player)`